src/plugins/nestedHeaders/stateManager/sourceSettings.js

```diff
@@ -0,0 +1,197 @@
+import { extend, isObject } from '../../../helpers/object';
+import { arrayEach } from '../../../helpers/array';
+import { normalizeSettings } from './settingsNormalizer';
+import { HEADER_CONFIGURABLE_PROPS } from './constants';
+
+/**
+ * The class manages and normalizes settings passed by the developer
+ * into the nested headers plugin. The SourceSettings class is a
+ * source of truth for tree builder (HeaderTree) module.
+ *
+ * @class {SourceSettings}
+ */
+export default class SourceSettings {
+  /**
+   * The normalized source data (normalized user-defined settings for nested headers).
+   *
+   * @private
+   * @type {Array[]}
+   */
+  #data = [];
+  /**
+   * The total length of the nested header layers.
+   *
+   * @private
+   * @type {number}
+   */
+  #dataLength = 0;
+  /**
+   * Columns count limit value trim source settings to that value. If columns
```

Comment:
Hm, I am not quite sure I get this sentence right but I would say: "Columns count limit value trim**s** "